src/Tribe/Views/V2/functions/classes.php

```diff
@@ -133,9 +134,9 @@ function month_day_classes( array $day, string $day_date, \DateTime $request_dat
 	$day_classes = [
 		'tribe-events-calendar-month__day'              => true,
 		// Add a class for the current day.
-		'tribe-events-calendar-month__day--current'     => $comparison_date->format( 'Y-m-d' ) === $day_date,
+		'tribe-events-calendar-month__day--current'     => $today_date === $day_date,
```

Comment:
The reverts an earlier fix.
The issue is, if the date is set (like we're viewing last month's calendar), `$today_date` still uses _today_ - so the classes will not be set based on the calendar date.

Reply:
Yes, but if I am looking at a specific date, should the classes reflect _today_ or the specific date?

Reply:
I believe the proper logic is spelled out in [Jira](https://stellarwp.atlassian.net/jira/software/c/projects/TEC/issues/TEC-4686?jql=project%20%3D%20%22TEC%22%20AND%20text%20~%20%22class%22%20ORDER%20BY%20created%20DESC).

Perhaps we need a filter here for folks that want a different behavior?

Reply:
I believe the intention was to have the classes based on the current date to help users target past and future events easily. I've asked for clarification from LG [here](https://stellarwp.atlassian.net/browse/TEC-4898?focusedCommentId=117723).

Reply:
Introduced: https://github.com/the-events-calendar/the-events-calendar/pull/3949
Ticket: https://stellarwp.atlassian.net/browse/TEC-4457

Reply:
On a side note, rather than bypassing the `$comparison_date` _completely_, we can alter the logic in the filter.

Reply:
@Camwyn I've made some tweaks based on your comments.